### PR TITLE
Clean up the handling of execution

### PR DESCRIPTION
**Pull request by blink1073**
The output area needs much of the raw information from the kernel messages to render output, clear output, handle response payloads, and stdin messages.  This PR modifies the `Kernel.IFuture` to provide a simpler interface that can be consumed by the output area.

### CODE REVIEW

**Comment by blink1073** [comment]
Note to self: implement https://github.com/jupyterlab/jupyterlab/issues/1227 in this PR if possible.

File: packages/cells/src/widget.ts [hunk 1]
```diff
@@ -643,6 +616,28 @@ namespace CodeCell {
     rendermime: RenderMime;
   }
 
+  /**
+   * Execute a cell given a client session.
+   */
+  export
+  function execute(cell: CodeCell, session: IClientSession): Promise<KernelMessage.IExecuteReplyMsg> {
```
Comment (ellisonbg) [inline packages/cells/src/widget.ts]: I like this how all turned out!

File: packages/cells/src/widget.ts [hunk 2]
```diff
@@ -643,6 +616,28 @@ namespace CodeCell {
...
+      model.executionCount = null;
+      model.outputs.clear();
+      return Promise.resolve(void 0);
+    }
+
+    model.executionCount = null;
+    cell.setPrompt('*');
+    model.trusted = true;
+
+    return OutputArea.execute(cell.outputArea, code, session).then(msg => {
```
Comment (ellisonbg) [inline packages/cells/src/widget.ts]: I have a very slight preference to swap the order of the `code` and `outputArea` arguments as it more closely matches how I think about the logic = "run this code and send the output to this output area."

File: packages/outputarea/src/widget.ts [hunk 3]
```diff
@@ -420,9 +425,28 @@ namespace OutputArea {
   }
 
   /**
+   * Execute code on an output area.
+   */
+  export
+  function execute(output: OutputArea, code: string, session: IClientSession): Promise<KernelMessage.IExecuteReplyMsg> {
```
Comment (ellisonbg) [inline packages/outputarea/src/widget.ts]: Maybe swap the order of code and output.

**Comment by ellisonbg** [comment]
Merging!

**Pull request merged**